From 3c6c98f44947cd90d510c21c570ccfcb3b4f0301 Mon Sep 17 00:00:00 2001
From: hscham <hscham@chromium.org>
Date: Thu, 17 Dec 2020 02:38:06 +0000
Subject: [PATCH] Make NamedValue hashable

For Python3 compatibility.
hash function added in crrev.com/c/2277687 for Field requires hash
function for EnumValue (self.default).

BUG=941669

Change-Id: I787bca64a05d379088a6e2846f03b76e258e0701
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2584728
Reviewed-by: Darin Fisher <darin@chromium.org>
Reviewed-by: Oksana Zhuravlova <oksamyt@chromium.org>
Commit-Queue: Grace Cham <hscham@chromium.org>
Cr-Commit-Position: refs/heads/master@{#837885}
---

diff --git a/mojo/public/tools/mojom/mojom/generate/module.py b/mojo/public/tools/mojom/mojom/generate/module.py
index 2f8e4d14..99ecd10 100644
--- a/mojo/public/tools/mojom/mojom/generate/module.py
+++ b/mojo/public/tools/mojom/mojom/generate/module.py
@@ -310,6 +310,9 @@
             and (self.parent_kind, self.mojom_name) == (rhs.parent_kind,
                                                         rhs.mojom_name))
 
+  def __hash__(self):
+    return hash((self.parent_kind, self.mojom_name))
+
 
 class BuiltinValue(object):
   def __init__(self, value):
